Backend/Tournament_User/TournamentUserMVSC/controllers/tournamentUserController.js

```diff
@@ -0,0 +1,459 @@
+const TournamentUserService = require('../services/tournamentUserService');  // Import service
+const TournamentService = require('../services/tournamentService');  // Import service
+
+
+/**
+ * Get participated tournament history for a user
+ * @param {Object} req.params - The request parameters TournamentID
+ * @param {string} req.params.UUID - The ID of the User.
+ * @returns {Json} Returns a list of tournament names.
+ * @throws {error} If UUID invalid.
+ */
+
+
+exports.getTournaments = async (req, res, next) => {
+    const { UUID } = req.params;
+
+    try {
+        let result =  await TournamentUserService.getTournaments(UUID);
+
+        // If no tournaments were found, return a 404 response
+        if (!result) {
+            throw new Error('No tournaments found');
+        }
+
+
+        // Extract names from the tournaments and put to a map.
+        const tournamentNames = result.map(tournament => tournament.name);
+
+
+
+        // Set the success response in res.locals.data and call next()
+        res.locals.data = {
+            statusCode: 200,
+            message: 'Tournament Found',
+            success: true,
+            content: tournamentNames,
+        };
+
+        next();
+
+    } catch (err) {
+
+        if(err.message === 'No tournaments found'){
+            err.statusCode = 404;
+        }
+
+        console.error('Error retrieving tournaments:', err);
+        next(err);  // Pass the error to the middleware
+    }
+};
+
+/**
+ * Get tournament matchups for a user
```

Comment:
think your comment param declarations and your code a bit don't match. Like your tournamentId is req.params not the request parameters, then UUID is req.body?